metadata/registry_test.go

```diff
@@ -2,12 +2,163 @@ package metadata
+
+			targetOrg:                  "exampleOrg2",
+			targetRepo:                 "exampleRepo2",
+			targetRefSpec:              "master",
+			targetRegistryRepoPath:     "",
+			targetRegistrySpecRepoPath: "registry.yaml",
+		},
+		{
+			uri: "github.com/exampleOrg3/exampleRepo3/tree/exampleBranch1",
+
+			targetOrg:                  "exampleOrg3",
+			targetRepo:                 "exampleRepo3",
+			targetRefSpec:              "exampleBranch1",
+			targetRegistryRepoPath:     "",
+			targetRegistrySpecRepoPath: "registry.yaml",
+		},
+		{
+			// Fails because `blob` refers to a file, but this refers to a directory.
+			uri:       "github.com/exampleOrg4/exampleRepo4/blob/master",
+			targetErr: errInvalidURI,
+		},
+		{
+			uri: "github.com/exampleOrg4/exampleRepo4/tree/exampleBranch2",
+
+			targetOrg:                  "exampleOrg4",
+			targetRepo:                 "exampleRepo4",
+			targetRefSpec:              "exampleBranch2",
+			targetRegistryRepoPath:     "",
+			targetRegistrySpecRepoPath: "registry.yaml",
+		},
+
+		//
+		// Parsing URIs with paths.
+		//
+		{
+			// Fails because referring to a directory requires a URI with
+			// `tree/{branchName}` prepending the path.
+			uri:       "github.com/exampleOrg6/exampleRepo6/path/to/some/registry",
+			targetErr: errInvalidURI,
+		},
+		{
+			uri: "github.com/exampleOrg5/exampleRepo5/tree/master/path/to/some/registry",
+
+			targetOrg:                  "exampleOrg5",
+			targetRepo:                 "exampleRepo5",
+			targetRefSpec:              "master",
+			targetRegistryRepoPath:     "path/to/some/registry",
+			targetRegistrySpecRepoPath: "path/to/some/registry/registry.yaml",
+		},
+		{
+			uri: "github.com/exampleOrg6/exampleRepo6/tree/exampleBranch3/path/to/some/registry",
+
+			targetOrg:                  "exampleOrg6",
+			targetRepo:                 "exampleRepo6",
+			targetRefSpec:              "exampleBranch3",
+			targetRegistryRepoPath:     "path/to/some/registry",
+			targetRegistrySpecRepoPath: "path/to/some/registry/registry.yaml",
+		},
+		{
+			// Fails because `blob` refers to a file, but this refers to a directory.
+			uri:       "github.com/exampleOrg7/exampleRepo7/blob/master",
+			targetErr: errInvalidURI,
+		},
+		{
+			// Fails because `blob` refers to a file, but this refers to a directory.
+			uri:       "github.com/exampleOrg5/exampleRepo5/blob/exampleBranch2",
+			targetErr: errInvalidURI,
+		},
+	}
+
+	for _, test := range tests {
+		// Make sure we correctly parse each URN as a bare-domain URI, as well as
+		// with 'http://' and 'https://' as prefixes.
+		for _, prefix := range []string{"http://", "https://", "http://www.", "https://www.", "www.", ""} {
+			// Make sure we correctly parse each URI even if it has the optional
+			// trailing `/` character.
+			for _, suffix := range []string{"/", ""} {
+				uri := prefix + test.uri + suffix
+
+				org, repo, refspec, registryRepoPath, registrySpecRepoPath, err := parseGitHubURI(uri)
```

Comment:
This function returns five values and an error. Are these values their own type?

Reply:
Nope.